packages/gatsby/src/utils/worker/child/render-html.ts

```diff
@@ -175,13 +195,38 @@ export const renderHTMLProd = async ({
           path: pagePath,
           unsafeBuiltinsUsageByPagePath,
         }
-        throw e
+
+        // If we're in Preview-mode, write out a simple error html file.
+        if (isPreview) {
+          const pageData = await readPageData(publicDir, pagePath)
+          const truncatedPageData = truncateObjStrings(pageData)
+
+          const html = `<h1>Preview build error</h1>
+        <p>There was an error when building the preview page for this page ("${pagePath}").</p>
+        <h3>Error</h3>
+        <pre><code>${e.stack}</code></pre>
+        <h3>Page component id</h3>
+        <p><code>${pageData.componentChunkName}</code></p>
+        <h3>Page data</h3>
+        <pre><code>${JSON.stringify(truncatedPageData, null, 4)}</code></pre>`
+
+          await fs.outputFile(generateHtmlPath(publicDir, pagePath), html)
+          previewErrors[pagePath] = {
```

Comment:
For very large sites we might be storing a ton of errors here right? I wonder if we should use a Map instead and check size before adding to it so we only add a certain number of errors. 
For ex if every page in a 1M page site had the same error for some reason, we would store 1M errors here right?

Reply:
it only adds the pagePath so the marginal extra information per page is pretty small. If this becomes a problem, we could e.g. add the first 100 pages with the error perhaps?

Reply:
On 2nd thought, yeah, just add the check now.

Reply:
Oh actually no — this is on the child process which is only handling 50 pages at a time. So the max number of errors would be 50 here. I'm not sure it makes a lot of sense to try to be more fancy.

Reply:
Nice, that makes sense 👍 thanks!